students/navdeep/session04/kata14.py

```diff
@@ -0,0 +1,39 @@
+import string
+
+def createTrigram(filename):
+    """
+    Function reads in the file data and creates a trigram. The punctuation is removed first
+    followed by the newlines.
+    :param filename: the file data we read in. Contains text
+    """
+    #Removes punctuation from text. Replaces all punctuation with a space
+    no_punctuation_data = str.maketrans(string.punctuation, ' '* len(string.punctuation))
+    new_file_data = filename.translate(no_punctuation_data)
+    #Replaces all carriage returns with a space
+    new_file_data = new_file_data.replace('\n', ' ')
```

Comment:
I think if you replace '\n' with an empty string ('\n', ''), you won't have to go back through and remove spaces. Take a look at the strip() method, too (you might find it easier to convert text to a list right away, for example, by using readlines from the file content).